discord_test.go

```diff
@@ -183,3 +184,120 @@ func TestRemoveHandler(t *testing.T) {
 		t.Fatalf("testHandler was not called once.")
 	}
 }
+
+func TestScheduledEvents(t *testing.T) {
+	if dgBot == nil {
+		t.Skip("Skipping, dgBot not set.")
+	}
+
+	beginAt := time.Now().Add(1 * time.Hour)
+	endAt := time.Now().Add(2 * time.Hour)
+	event, err := dgBot.GuildScheduledEventCreate(envGuild, &GuildScheduledEventParams{
+		Name:               "Test Event",
+		PrivacyLevel:       GuildScheduledEventPrivacyLevelGuildOnly,
+		ScheduledStartTime: &beginAt,
+		ScheduledEndTime:   &endAt,
+		Description:        "Awesome Test Event created on livestream",
+		EntityType:         GuildScheduledEventEntityTypeExternal,
+		EntityMetadata: &GuildScheduledEventEntityMetadata{
+			Location: "https://discord.com",
+		},
+	})
+	defer dgBot.GuildScheduledEventDelete(envGuild, event.ID)
+
+	if err != nil || event.Name != "Test Event" {
+		t.Fatal(err)
+	}
+
+	events, err := dgBot.GuildScheduledEvents(envGuild, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var foundEvent *GuildScheduledEvent
+	for _, e := range events {
+		if e.ID == event.ID {
+			foundEvent = e
+			break
+		}
+	}
+	if foundEvent.Name != event.Name {
+		t.Fatal("err on GuildScheduledEvents endpoint. Missing Scheduled Event")
+	}
+
+	getEvent, err := dgBot.GuildScheduledEvent(envGuild, event.ID, true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if getEvent.Name != event.Name {
+		t.Fatal("err on GuildScheduledEvent endpoint. Mismatched Scheduled Event")
+	}
+
+	eventUpdated, err := dgBot.GuildScheduledEventEdit(envGuild, event.ID, &GuildScheduledEventParams{Name: "Test Event Updated"})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if eventUpdated.Name != "Test Event Updated" {
+		t.Fatal("err on GuildScheduledEventUpdate endpoint. Scheduled Event Name mismatch")
+	}
+
+	users, err := dgBot.GuildScheduledEventUsers(envGuild, event.ID, 1, true, "1", "1")
```

Comment:
Are you sure it shouldn't be `"", ""`?

Reply:
It's to force the condition `beforeID` and `afterID` to run during the test. As no user is registered to the event during the test, it is impossible to fill in consistent IDs. It would be necessary to have a test user and to make him register to the event but it is currently impossible via the API

Reply:
> It's to force the condition `beforeID` and `afterID` to run during the test. As no user is registered to the event during the test, it is impossible to fill in consistent IDs. It would be necessary to have a test user and to make him register to the event but it is currently impossible via the API

I meant, why not omit the parameters, since we don't need to get all of the event users

Reply:
Just the philosophy of run all branch in function without crashes. When Discord add possibility to register an event via API, I will add Endpoints and tests suite without correct value and omit here (because when this is the case all branches of the function can be tested without false values)

But, if you don't like this philosophy, I can omit it and add comment on test to explain why and when we need to add this test suit

Reply:
Would be great if you add the comment explaining why it's `1`. But for me, this doesn't really make sense, since you would probably always get zero users, since it would be `id < 1 && id > 1` which means condition will be always false